spec-tests/src/test/java/org/candlepin/spec/content/SkuLevelEnableOverrideSpecTest.java

```diff
@@ -0,0 +1,272 @@
+        ProductDTO product = Products.randomEng()
+            .addAttributesItem(ProductAttributes.ContentOverrideDisabled.withValue(content1.getId()))
+            .providedProducts(Set.of(providedProduct));
+        product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+
+        PoolDTO pool = ownerClient.createPool(owner.getKey(), Pools.random(product));
+        consumerClient.consumers().bindPool(consumer.getUuid(), pool.getId(), 1);
+
+        EntitlementDTO ent = consumerClient.consumers().listEntitlements(consumer.getUuid()).get(0);
+        String value = CertificateUtil.standardExtensionValueFromCert(
+            ent.getCertificates().iterator().next().getCert(),
+            "1.3.6.1.4.1.2312.9.2." + content1.getId() + ".1.8");
+        assertEquals("0", value);
+    }
+
+    @Test
+    public void shouldSkuOverrideForEnabledSupercededByEnvironmentPromotionV1() {
+        OwnerDTO owner = ownerClient.createOwner(Owners.random());
+        EnvironmentDTO env = ownerClient.createEnv(owner.getKey(), Environments.random());
+        ConsumerDTO consumer = Consumers.random(owner)
+            .facts(Map.of("system.certificate_version", "1.0"))
+            .environments(List.of(env));
+        consumer = consumerClient.createConsumer(consumer);
+        assertThat(!consumer.getEnvironments().isEmpty());
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        ProductDTO providedProduct = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.randomEng());
+        ContentDTO content1 = ownerContentApi.createContent(owner.getKey(), Contents.random());
+        ownerProductApi.addContent(owner.getKey(), providedProduct.getId(), content1.getId(), false);
+        ProductDTO product = Products.randomEng()
+            .addAttributesItem(ProductAttributes.ContentOverrideDisabled.withValue(content1.getId()))
+            .providedProducts(Set.of(providedProduct));
+        product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+
+        PoolDTO pool = ownerClient.createPool(owner.getKey(), Pools.random(product));
+
+        // Override enabled to true:
+        ContentToPromoteDTO toPromote = new ContentToPromoteDTO()
+            .contentId(content1.getId())
+            .enabled(true);
+        AsyncJobStatusDTO job = client.environments().promoteContent(
+            env.getId(), List.of(toPromote), true);
+        job = client.jobs().waitForJob(job.getId());
+        assertThatJob(job).isFinished();
+
+        consumerClient.consumers().bindPool(consumer.getUuid(), pool.getId(), 1);
+        EntitlementDTO ent = consumerClient.consumers().listEntitlements(consumer.getUuid()).get(0);
+        String value = CertificateUtil.standardExtensionValueFromCert(
+            ent.getCertificates().iterator().next().getCert(),
+            "1.3.6.1.4.1.2312.9.2." + content1.getId() + ".1.8");
```

Comment:
There is a custom assertion that simplifies work with certs.

```java
assertThatCert(X509Cert.fromEnt(ent))
    .hasContentRepoEnabled(content1);
```
`fromEnt` currently accepts only JsonNode so you'll need to add an overloaded version.